src/box/vy_tx.c

```diff
@@ -1157,10 +1151,13 @@ vy_tx_manager_abort_writers_for_ddl(struct vy_tx_manager *xm,
 			*need_wal_sync = true;
 		if (tx->state != VINYL_TX_READY)
 			continue;
-		if (tx->last_stmt_space == space ||
-		    write_set_search_key(&tx->write_set, lsm,
-					 lsm->env->empty_key) != NULL)
-			vy_tx_abort(tx);
+		struct txn_stmt *stmt;
```

Comment:
Offtop: `wal_sync()` seems insufficient here - prepared transactions can be aborted even after reaching WAL due to replication failure.

Reply:
Talked f2f. Turns out there's a bug here if sync replication is enabled: `wal_sync()` doesn't wait for quorum confirmation so it doesn't guarantee that all pending statements are committed or rolled back. Looks like we need to wait for limbo resolution here as well. This is another issue that will be addressed separately though.